modules/internal/localeModels/gpu/LocaleModel.chpl

```diff
@@ -30,10 +30,167 @@
 module LocaleModel { 
 
   public use LocaleModelHelpGPU;
-  public use LocaleModelHelpMem;
   require "-lcudart";
 
-  use IO;
+  use IO, SysCTypes, CPtr;
+
+  private inline
+  proc runningOnGPUSublocale(): bool {
```

Comment:
Later on, when we call `chpl_gpu_init` implicitly, I'm not so sure we can check for a GPU context in this way.

Instead, I would expect it to only consider the sublocale.

Does this PR work if this function is just `chpl_task_getRequestedSubloc()>0` ?

Reply:
> Does this PR work if this function is just chpl_task_getRequestedSubloc()>0 ?

I agree that this'll be a problem when we initialize GPU context implicitly.

In my initial attempt where GPU sublocale allocations were made in the non-unified memory, doing what you suggest was problematic. When setting up the locales, you'd want the GPU sublocale instance to be on the CPU. That's why the logic currently checks if there's a GPU context (which you won't have during locale setup, resulting in all the allocations during setup be on the CPU).

With the use of unified memory, we may be able to get away with dropping that context check.

Reply:
Ah.. that issue is still there. Because if you don't have a context, you cannot allocate unified memory, because you cannot use any cuda function. I'll look at this while working on making `chpl_gpu_init` calls implicit. (https://github.com/Cray/chapel-private/issues/2377)